daemon/mgr/container_envs.go

```diff
@@ -24,110 +20,40 @@ import (
-
-	return nil
-}
-
-// getLocalEnv gets local ENV from specified env shell files.
-// Currently this is file /etc/profile.d/pouchenv.sh in the container.
-// And every row of this file has a format of `export AAA="BBB"`.
-func getLocalEnv(filename string) (map[string]string, error) {
-	envMap := make(map[string]string)
-
-	b, err := ioutil.ReadFile(filename)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read env file %s: %v", filename, err)
-	}
-
-	rawData := string(b)
-	envsh := strings.Trim(rawData, "\n")
-	envs := strings.Split(envsh, "\n")
-	for _, env := range envs {
-		env = strings.TrimPrefix(env, "export ")
-		key, value, err := utils.ConvertStrToKV(env)
-		if err != nil {
-			// TODO: just ignore the error or return err?
-			return envMap, err
-		}
-		// FIXME: do we need to string.Trim(value, "\"")
-		envMap[key] = value
-	}
-
-	return envMap, nil
-}
-
-// combineLocalAndInputEnv combines local envs extracted from pouchenv.sh and input envs.
-// traverse env(key, value) from local file, and replace it if key also exists in input env.
-// traverse env(key, value) from input env, if it does not exist in local env, just append it.
-// FIXME: why not just only traverse input env?
-func combineLocalAndInputEnv(inputEnv, localEnv map[string]string) map[string]string {
-	for key, valueInLocal := range localEnv {
-		valueInInput, exist := inputEnv[key]
-		if !exist {
-			// if key in the local env does not exist in input env,
-			// it means this key will never be updated by input env.
-			continue
-		}
-		// key in local env exists in input env.
-		// TODO: explain this part by ziren
-		s := strings.Replace(valueInInput, "\"", "\\\"", -1)
-		valueInInput = strings.Replace(s, "$", "\\$", -1)
-
-		if key == "PATH" {
-			// FIXME: how to do if there is already a substr of `:$PATH` in itself.
-			valueInInput = valueInInput + ":$PATH"
-		}
-		// two values are not equal, use the input one.
-		if valueInInput != valueInLocal {
-			logrus.Infof("env exists and the value is not same, key=%s, old value=%s, new value=%s", key, valueInLocal, valueInInput)
-			localEnv[key] = valueInInput
+	var (
+		str string
+	)
+	for _, kv := range env {
+		parts := strings.SplitN(kv, "=", 2)
+		if len(parts) == 1 {
+			parts = append(parts, "")
 		}
-	}
-	// append the brand new input env to local env map.
-	for key, value := range inputEnv {
-		if _, ok := localEnv[key]; !ok {
-			logrus.Infof("env does not exist, set new key value pair, new key=%s, new value=%s", key, value)
-			localEnv[key] = value
+		if len(parts[1]) > 0 && !strings.Contains(parts[0], ".") {
+			s := strings.Replace(parts[1], "\\", "\\\\", -1)
+			s = strings.Replace(s, "\"", "\\\"", -1)
+			s = strings.Replace(s, "$", "\\$", -1)
+			if parts[0] == "PATH" {
+				s = parts[1] + ":$PATH"
+			}
+			str += fmt.Sprintf("export %s=\"%s\"\n", parts[0], s)
 		}
 	}
-	return localEnv
+	ioutil.WriteFile(filepath.Join(baseFs, pouchEnvFile), []byte(str), 0755)
```

Comment:
OK, i got you, we will just write the env to the file, so we just need check the existence of Dir